libs/npm/core/src/utils/elastic/generateQueryValues.ts

```diff
@@ -25,11 +25,23 @@ export const generateQueryValues = (
   const values = [
     generatePublishedOnQuery(settings),
     generatePostedOnQuery(settings),
-    generateTerms('sourceId', settings.sourceIds),
-    generateTerms('mediaTypeId', settings.mediaTypeIds),
-    generateTerms('seriesId', settings.seriesIds),
-    generateTerms('contributorId', settings.contributorIds),
-    generateTerms('contentType', settings.contentTypes),
+    generateTerms('sourceId', settings.sourceIds, settings.emptyArrayTerms?.includes('sourceId')),
```

Comment:
Why can you not do something simple like this on line 68?

`].filter((v) => (v !== undefined && !Array.isArray(v)) || (Array.isArray(v) && v.length > 0));`

Reply:
I can push out a test to see how this line behaves in DEV.

Reply:
Because this on line 16 of generateTerms.ts:
return fieldValues.length > 0
    ? {
        terms: { [field]: fieldValues },
      }
    : undefined;

Would detect an Zero length array, and remove it from the ES call.

Reply:
> I can push out a test to see how this line behaves in DEV.

We can make it on Monday morning, no need to rush. I'm sure it will work in dev, but we can re think a better approach. Because it really seems a little complicated from outsiders.

Reply:
I simply don't know why we would pass an empty array to elastic

Reply:
> I simply don't know why we would pass an empty array to elastic

Me neither. :(